lib/ansible/plugins/inventory/vmware.py

```diff
@@ -0,0 +1,238 @@
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: vmware
+    plugin_type: inventory
+    short_description: vmware inventory source
+    description:
+        - Get inventory hosts from VMware environment.
+        - Uses any file which ends with vmware.yml or vmware.yaml as a YAML configuration file.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    options:
+        running_only:
+            description: toggles showing all vms vs only those currently running
+            type: boolean
+            default: False
+        hostname:
+            description: Name of vCenter or ESXi server
+            env:
+              - name: VMWARE_SERVER
+        username:
+            description: Name of vSphere admin user
+            env:
+              - name: VMWARE_USERNAME
+        password:
+            description: Password of vSphere admin user
+            env:
+              - name: VMWARE_PASSWORD
+        port:
+            description: Port number used to connect to vCenter or ESXi Server
+            env:
+              - name: VMWARE_PORT
+        validate_certs:
+            description: a
+            default: True
+            type: boolean
+'''
+
+EXAMPLES = '''#
+'''
+
+import ssl
+import atexit
+
+try:
+    # requests is required for exception handling of the ConnectionError
+    import requests
```

Comment:
if using requests you really need to specify version, the api varies a LOT across versions, mainly why we recommend using open_url/fetch_url instead

Reply:
`requests` is just used for handling `ConnectionError` and Session management.

Reply:
still, versioning is a problem, the 'default installed version' varies widely across distros/os versions.